airlock/views/workspace.py

```diff
@@ -140,40 +144,102 @@ def workspace_contents(request, workspace_name: str, path: str):
     return serve_file(request, renderer)
 
 
+@instrument(func_attributes={"workspace": "workspace_name"})
+@require_http_methods(["POST"])
+def workspace_multiselect(request, workspace_name: str):
+    """User has selected multiple files and wishes to perform an action on them.
+
+
+    In some cases, we need to return a further form as a modal for more information for each file.
+    In others, we can just perform the requested action and redirect.
+    """
+    workspace = get_workspace_or_raise(request.user, workspace_name)
+
+    multiform = MultiselectForm(request.POST)
+
+    if multiform.is_valid():
+        action = multiform.cleaned_data["action"]
+
+        if action == "add_files":
+            add_files_form = AddFilesForm(
+                release_request=bll.get_current_request(workspace_name, request.user),
+                files=multiform.cleaned_data["selected"],
+            )
+            # this was the only way I could find to inject a value into
+            # initial, which is used in the form rendering
+            add_files_form.fields["next_url"].initial = multiform.cleaned_data[
+                "next_url"
+            ]
+
+            return TemplateResponse(
+                request,
+                template="add_files.html",
+                context={
+                    "form": add_files_form,
+                    "add_file_url": reverse(
+                        "workspace_add_file",
+                        kwargs={"workspace_name": workspace_name},
+                    ),
+                },
+            )
+        # TODO: withdraw action
+        else:
+            raise Http404(f"Invalid action {action}")
+    else:
+        display_form_errors(request, multiform)
+        return redirect(multiform.cleaned_data.get("next_url", workspace.get_url()))
+
+
 @instrument(func_attributes={"workspace": "workspace_name"})
 @require_http_methods(["POST"])
 def workspace_add_file_to_request(request, workspace_name):
     workspace = get_workspace_or_raise(request.user, workspace_name)
-    relpath = UrlPath(request.POST["path"])
+
+    # pull out list of files from raw POST data to be able to create the form with
+    files = [v for k, v in request.POST.items() if k.startswith("file_")]
+    # get raw source of POST, to use in case form doesn't validate
+    next_url = request.POST["next_url"]
```

Comment:
Use `.get()` and a default next url just in case it's missing?

Reply:
Yeah, I've reworked this to more formal.